plugin/pulp/plugin/importer.py

```diff
@@ -0,0 +1,18 @@
+from pulp.platform.models import RepositoryImporter
+
+
+class RepositoryImporter(RepositoryImporter):
+    """
+    The base RepositoryImporter object
+    """
+
+    def sync(self):
+        """
+        Perform a sync
```

Comment:
That's a fair point. The broader context is that I was initially surprised to see that we'd have a method on a model that creates and/or changes state in lots of other models. It feels more like controller code, and inappropriate as a model method. But having talked through it with @bmbouter , the reasoning for doing it this way is to simplify the plugin writer experience, and simplify the platform's discovery process.

If we ask plugin writers to make a subclass of a separate ``Importer`` class of some kind, that would add complexity for them, and we would need a way to discover that class. (we have a nice simple way to discover the models that is outlined elsewhere) This design is a compromise to make the connection between the platform and the plugin's workflow simple, easy, and discoverable.

All of that said, my suggestion here is in large part to explain some of the above in the doc block, so that plugin writers and other readers understand why we have a method on a model class that is going to change state with lots of other models. I think a small hint to the plugin writer is appropriate, lest they think we are trying to lead them down the path of putting their sync code here.